### PR TITLE
Backport ES2K changes to ipdk-io

### PR DESCRIPTION
**Pull request by ffoulkes**
Open-source ES2K support

### CODE REVIEW

File: stratum/pipelines/ptf/scapy_exec.bzl
Comment (nupuruttarwar) [inline stratum/pipelines/ptf/scapy_exec.bzl:1]: Are we removing files under stratum/pipelines/ since latest stratum doesn't have this? 
Are we rebasing the code to latest stratum as well as part of this PR

File: stratum/lib/utils.cc
Comment (nupuruttarwar) [inline stratum/lib/utils.cc:274]: Where are we using this function?
I understood from Derek that this is to transport the encryption key via Protobuf for IPSEC. If it's specific to how we transport the key for ES2ky, then we should introduce a functionsn in utils module under tdi instead of modifying common stratum lib utils

File: stratum/hal/lib/yang/yang_parse_tree_paths.h
Comment (nupuruttarwar) [inline stratum/hal/lib/yang/yang_parse_tree_paths.h:55]: for IPSEC?

File: stratum/hal/lib/yang/yang_parse_tree.h
Comment (nupuruttarwar) [inline stratum/hal/lib/yang/yang_parse_tree.h:243]: will be executed in case node is deleted?

File: stratum/hal/lib/tdi/tdi_lut_manager.cc
Comment (nupuruttarwar) [inline stratum/hal/lib/tdi/tdi_lut_manager.cc:65]: line 65 and 69 can be removed

File: stratum/hal/lib/tdi/tdi_constants.h
Comment (nupuruttarwar) [inline stratum/hal/lib/tdi/tdi_constants.h:20]: Will changing these values affect backward compatibility?

File: stratum/hal/bin/tdi/es2k/es2k_skip_p4.conf
Comment (nupuruttarwar) [inline stratum/hal/bin/tdi/es2k/es2k_skip_p4.conf:5]: changes to support e2100 value are already merged, we can change this to e2100

File: stratum/hal/lib/tdi/es2k/es2k_port_manager.cc
Comment (nupuruttarwar) [inline stratum/hal/lib/tdi/es2k/es2k_port_manager.cc:141]: We don't add port for ES2K. we can keep the function and return OK. Function on line 146 can be removed for ES2K

File: stratum/lib/utils.cc
Comment (ffoulkes) [reply stratum/lib/utils.cc:274]: This function is only used in `TdiIpsecManager`. I'd suggest that the function be defined there.

**Review by 5abeel — APPROVED** [review]
LGTM

File: stratum/pipelines/ptf/scapy_exec.bzl
Comment (ffoulkes) [reply stratum/pipelines/ptf/scapy_exec.bzl:1]: Fixing the Bazel build required, among other things, that we update the version of gRPC used in the Bazel build. This broke other parts of the build. The most straightforward solution to the cascade of issues was to cherry-pick the dependencies and their supporting changes from the latest version of Stratum.

One of the issues in the cascade was that we could no longer build the Python files under `stratum/pipelines`. The latest version of stratum removed these files and their BUILD rules, so I did likewise. Problem solved, and we are one step closer to updating to the latest version of stratum.

We are not rebasing the code to the latest version of Stratum. That's still in our future.

File: stratum/hal/bin/tdi/es2k/es2k_skip_p4.conf
Comment (ffoulkes) [reply stratum/hal/bin/tdi/es2k/es2k_skip_p4.conf:5]: The instructions from Dan are to wait until after the merge.

**Pull request merged**
